app/Conference.php

```diff
@@ -49,4 +56,49 @@ public function onlineFriends()
     {
         return $this->hasMany(Friend::class)->where('type', 'online');
     }
+
+    public function makeIntroduction($username)
+    {
+        $friend = new Friend([
```

Comment:
It feels like the amount of switching we're doing based on `$this->isUpcoming()` is revealing some sort of data structure issue still.

But I can't figure out what we could change or remove.